src/trix/models/composition.coffee

```diff
@@ -109,8 +120,14 @@ class Trix.Composition extends Trix.BasicObject
       else
         if block.isEmpty()
           @removeLastBlockAttribute()
-        else if block.text.getStringAtRange([endLocation.offset - 1, endLocation.offset]) is "\n"
+        else if previousCharacter is "\n"
           @breakFormattedBlock()
+        else if breaksOnReturn and nextCharacter is "\n"
+          @breakFormattedBlock()
+        else if breaksOnReturn and startLocation.offset is 0
+          @insertBlockBreak()
+        else if breaksOnReturn
+          @insertBlockBreak()
```

Comment:
Doesn't look like these all need to be separate conditions. How about combing the ones that perform the same action with an `or`?